src/effects/PerTrackEffect.h

```diff
@@ -402,4 +402,33 @@ class EffectStage final : public AudioGraph::Source {
    bool mLatencyDone{ false };
    bool mCleared{ false };
 };
+
+namespace AudioGraph{
+//! Copies from a Source to a Sink, mediated by Buffers
+struct Task {
```

Comment:
Should it be final?

Reply:
For now it can be.  There might be future need to make this an abstract base class, if we figure out multi-core.  And what is now in EffectStage::Produce would instead become another Task viewing its upstream and down as Source and Sink.

But I haven't arrived at those details yet.